### PR TITLE
Init metrics for the dual writer

### PR DESCRIPTION
**Pull request by leonorfmartins**
**What is this feature?**

Inits dual_writer metrics

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/search-and-storage-team/issues/41

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/apiserver/standalone/factory.go [hunk 1]
```diff
@@ -86,8 +86,8 @@ func (p *DummyAPIFactory) MakeAPIServer(_ context.Context, tracer tracing.Tracer
 				Client: client.NewTestDataClient(),
 			},
 			client.NewTestDataRegistry(),
-			nil,                      // legacy lookup
-			prometheus.NewRegistry(), // ???
+			nil, // legacy lookup
+			prometheus.DefaultRegisterer,
```
Comment (marefr) [inline pkg/services/apiserver/standalone/factory.go:90]: With import of pkg/infra/metrics

```suggestion
			metrics.ProvideRegisterer(),
```

File: pkg/apiserver/rest/metrics.go [hunk 2]
```diff
@@ -37,10 +38,17 @@ var DualWriterOutcome = prometheus.NewHistogramVec(prometheus.HistogramOpts{
 	NativeHistogramBucketFactor: 1.1,
 }, []string{"mode", "name", "method"})
 
-func (m *dualWriterMetrics) init() {
+func (m *dualWriterMetrics) init(log logr.Logger) {
 	m.legacy = DualWriterLegacyDuration
 	m.storage = DualWriterStorageDuration
 	m.outcome = DualWriterOutcome
+	reg := prometheus.DefaultRegisterer
```
Comment (marefr) [inline pkg/apiserver/rest/metrics.go:45]: With import of pkg/infra/metrics

```suggestion
	reg := metrics.ProvideRegisterer()
```

**Review by marefr — COMMENTED** [review]
Some suggestions. Any problem using `metrics.ProvideRegisterer()`?

File: pkg/apiserver/rest/metrics.go (hunk 2 above)
Comment (leonorfmartins) [reply pkg/apiserver/rest/metrics.go:45]: there's a linting rule disallowing imports from grafana to the apiserver pkg, so I'd rather keep what we have here.

Comment (marefr) [reply pkg/apiserver/rest/metrics.go:45]: Then take a prometheus.Registerer as a dependency/requirement for instantiating `dualWriterMetrics`?

Comment (leonorfmartins) [reply pkg/apiserver/rest/metrics.go:45]: that was actually my first attempt (see [PR](https://github.com/grafana/grafana/pull/89003)), but I guess it's way more straightforward if I just use the default prometheus register, which it seems to work as a singleton for the whole application, than passing the register around for every entity

**Comment by leonorfmartins** [comment]
closing this in favour of https://github.com/grafana/grafana/pull/89003 because, we cant just use a default register:
- we could in theory use the grafana one (living in `infra/metrics`) but due to linting rules and the principle of wanting to keep grafana and apiserver completely separate in terms of dependencies, we don't want to do that
- the advantage of using the grafana register one is that it doesn't use the default registerer but rather a new registry. This is important, because using the default one prevents k8s metrics from being exposed.

We can't also just create a different registry service under `apiserver` because it has 2 operating modes: embedded to grafana and standalone. The service metrics would work on the standalone mode, but not in the embedded one, as it would mean grafana would have 2 registries, which we want to avoid.

**Pull request closed without merging**